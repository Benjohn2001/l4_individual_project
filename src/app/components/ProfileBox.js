import React from "react";
import PropTypes from "prop-types";
import { Text, View } from "react-native";

function ProfileBox(props) {
  const { name, username, bio } = props;
  return (
    <View className="py-3 w-full h-7/12 items-center mx-1.5 bg-secondaryPurple rounded-2xl">
      <Text className="text-black text-2xl py-1 text-center mx-3">{name}</Text>
      <Text className="text-black text-xl py-1 text-center mx-3">
        @{username}
      </Text>
      <Text className="text-black text-xl py-1 text-center mx-3">{bio}</Text>
    </View>
  );
}

ProfileBox.propTypes = {
  name: PropTypes.string.isRequired,
  username: PropTypes.string.isRequired,
  bio: PropTypes.string.isRequired,
};

export default ProfileBox;
