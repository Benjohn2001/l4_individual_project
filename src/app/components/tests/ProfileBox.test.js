import renderer from "react-test-renderer";
import ProfileBox from '../ProfileBox';

test('Check ProfileBox renders correctly',()=>{
      const domTree = renderer.create(
        <ProfileBox 
            name="joe"
            username="jb"
            bio="joeb"
        />
        ).toJSON();
      expect(domTree).toMatchSnapshot();
})