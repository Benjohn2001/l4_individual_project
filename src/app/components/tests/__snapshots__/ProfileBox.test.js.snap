// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check ProfileBox renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "paddingBottom": 12,
        "paddingTop": 12,
      },
      Object {
        "width": "100%",
      },
      Object {
        "alignItems": "center",
      },
      Object {
        "marginLeft": 6,
        "marginRight": 6,
      },
      Object {
        "backgroundColor": "#E7E7FF",
      },
      Object {
        "borderBottomLeftRadius": 16,
        "borderBottomRightRadius": 16,
        "borderTopLeftRadius": 16,
        "borderTopRightRadius": 16,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        Object {
          "fontSize": 24,
          "lineHeight": 32,
        },
        Object {
          "paddingBottom": 4,
          "paddingTop": 4,
        },
        Object {
          "textAlign": "center",
        },
        Object {
          "marginLeft": 12,
          "marginRight": 12,
        },
      ]
    }
  >
    joe
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        Object {
          "fontSize": 20,
          "lineHeight": 28,
        },
        Object {
          "paddingBottom": 4,
          "paddingTop": 4,
        },
        Object {
          "textAlign": "center",
        },
        Object {
          "marginLeft": 12,
          "marginRight": 12,
        },
      ]
    }
  >
    @
    jb
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        Object {
          "fontSize": 20,
          "lineHeight": 28,
        },
        Object {
          "paddingBottom": 4,
          "paddingTop": 4,
        },
        Object {
          "textAlign": "center",
        },
        Object {
          "marginLeft": 12,
          "marginRight": 12,
        },
      ]
    }
  >
    joeb
  </Text>
</View>
`;
